traefik/oam/register.go

```diff
@@ -117,3 +125,93 @@ func load(basePath string) ([]schemaDefinitionPathSet, error) {
 
 	return res, nil
 }
+
+//RegisterWorkLoadsDynamically ...
+func RegisterWorkLoadsDynamically(runtime, host string) error {
+	av, cv, err := getLatestValidAppVersionAndChartVersion()
+	if err != nil {
+		return err
+	}
+	fmt.Println("version ", av)
+	m := manifests.Config{
+		Name:        "Traefik",
+		MeshVersion: av,
+		Filter: manifests.CrdFilter{
+			RootFilter:    []string{"$[?(@.kind==\"CustomResourceDefinition\")]"},
+			NameFilter:    []string{"$..[\"spec\"][\"names\"][\"kind\"]"},
+			VersionFilter: []string{"$..spec.versions[0]", " --o-filter", "$[0]"},
+			GroupFilter:   []string{"$..spec", " --o-filter", "$[]"},
+			SpecFilter:    []string{"$..openAPIV3Schema.properties.spec", " --o-filter", "$[]"},
+		},
+	}
+	url := "https://helm.traefik.io/mesh/maesh-" + cv + ".tgz"
+	comp, err := manifests.GetFromHelm(url, manifests.SERVICE_MESH, m)
+	if err != nil {
+		return err
+	}
+	for i, def := range comp.Definitions {
+
+		var ord adapter.OAMRegistrantData
+		ord.OAMRefSchema = comp.Schemas[i]
+
+		//Marshalling the stringified json
+		ord.Host = host
+		definitionMap := map[string]interface{}{}
+		if err := json.Unmarshal([]byte(def), &definitionMap); err != nil {
+			return err
+		}
+		// To be shifted in meshkit
+		definitionMap["apiVersion"] = "core.oam.dev/v1alpha1"
+		definitionMap["kind"] = "WorkloadDefinition"
+		ord.OAMDefinition = definitionMap
+		ord.Metadata = map[string]string{
+			config.OAMAdapterNameMetadataKey: config.TraefikMeshOperation,
+		}
+		// send request to the register
+		backoffOpt := backoff.NewExponentialBackOff()
+		backoffOpt.MaxElapsedTime = 10 * time.Minute
```

Comment:
This routine is common to all adapters. It seems that this would be defined in Meshery Adapter Library, @utkarsh-pro.

The timeout should be externally configurable.